taichi/transforms/lower_access.cpp

```diff
@@ -83,8 +85,13 @@ class LowerAccess : public IRVisitor {
     Stmt *last = lowered.push_back<GetRootStmt>();
 
     int path_inc = int(snode_op != SNodeOpType::undefined);
-    for (int i = 0; i < (int)snodes.size() - 1 + path_inc; i++) {
+    int length = (int)snodes.size() - 1 + path_inc;
+    for (int i = 0; i < length; i++) {
       auto snode = snodes[i];
+      if (is_bit_vectorized && snode->type == SNodeType::bit_array &&
+          i == length - 1 && snodes[i - 1]->type == SNodeType::dense) {
+        continue;
+      }
```

Comment:
core change: do not generate lookup/getch for bit_array snode when it's the last snode and its parent is a dense node.